pkg/auto/release.go

```
@@ -0,0 +1,181 @@
+	r := &Release{
+		git:           d.Git,
+		VR:            d.VR,
+		ChartVersion:  v,
+		Chart:         c,
+		ForkRemoteURL: f,
+	}
+
+	var ok bool
+	var assetVersions []lifecycle.Asset
+
+	assetVersions, ok = s.AssetsToBeReleased[r.Chart]
+	if !ok {
+		assetVersions, ok = s.AssetsToBeForwardPorted[r.Chart]
+		if !ok {
+			return nil, errors.New("no asset version to release for chart:" + r.Chart)
+		}
+	}
+
+	var assetVersion string
+	for _, version := range assetVersions {
+		if version.Version == r.ChartVersion {
+			assetVersion = version.Version
+			break
+		}
+	}
+	if assetVersion == "" {
+		return nil, errors.New("no asset version to release for chart:" + r.Chart + " version:" + r.ChartVersion)
+	}
+
+	r.AssetPath, r.AssetTgz = mountAssetVersionPath(r.Chart, assetVersion)
+
+	// Check again if the asset was already released in the local repository
+	if exist := checkAssetReleased(r.AssetPath); exist {
+		return nil, errors.New("asset already released for chart:" + r.Chart + " version:" + r.ChartVersion)
+	}
+
+	// Check if we have a release.yaml file in the expected path
+	if exist, err := filesystem.PathExists(d.RootFs, path.RepositoryReleaseYaml); err != nil || !exist {
+		return nil, errors.New("release.yaml not found")
+	}
+
+	r.ReleaseYamlPath = filesystem.GetAbsPath(d.RootFs, path.RepositoryReleaseYaml)
+
+	return r, nil
+}
+
+// PullAsset will execute the release porting for a chart in the repository
+func (r *Release) PullAsset() error {
+	if err := r.git.FetchBranch(r.VR.DevBranch); err != nil {
+		return err
+	}
+
+	if err := r.git.CheckFileExists(r.AssetPath, r.VR.DevBranch); err != nil {
+		return fmt.Errorf("asset version not found in dev branch: %w", err)
+	}
+
+	if err := r.git.CheckoutFile(r.VR.DevBranch, r.AssetPath); err != nil {
+		return err
+	}
+
+	if err := r.git.ResetHEAD(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkAssetReleased(chartVersion string) bool {
```

Let's change this from bool to error. We shouldn't let errors get ignored.